internal/lightcone/nihility/reforgedremembrance/reforgedremembrance.go

```diff
@@ -0,0 +1,102 @@
+package reforgedremembrance
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	remembrance key.Modifier = "reforged-remembrance " // rememberance = Prophet stack if i do this correctly
+	atkBuff     key.Modifier = "reforged-remembrance-atk-buff"
+	defShred    key.Modifier = "reforged-remembrance-def-shred"
+)
+
+type state struct {
+	atkBuff, defShred float64
+}
+
+// Increases the wearer's Effect Hit Rate by 40%. When the wearer deals DMG to an enemy
+// inflicted with Wind Shear, Burn, Shock, or Bleed, each respectively grants 1 stack of Prophet,
+// stacking up to 4 time(s). In a single battle, only 1 stack of Prophet can be granted for each
+// type of DoT. Every stack of Prophet increases wearer's ATK by 5% and enables the DoT dealt
+// to ignore 7.2% of the target's DEF.
+func init() {
+	lightcone.Register(key.ReforgedRemembrance, lightcone.Config{
+		CreatePassive: Create,
+		Rarity:        5,
+		Path:          model.Path_NIHILITY,
+		Promotions:    promotions,
+	})
+	modifier.Register(remembrance, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnAfterHit: addProphetStack,
+		},
+	})
+	modifier.Register(atkBuff, modifier.Config{
+		Stacking:          modifier.ReplaceBySource,
+		StatusType:        model.StatusType_STATUS_BUFF,
+		MaxCount:          4,
+		CountAddWhenStack: 1,
+		Listeners: modifier.Listeners{
+			OnAdd: recalcAtkBuff,
+		},
+	})
+	modifier.Register(defShred, modifier.Config{
+		BehaviorFlags: []model.BehaviorFlag{
+			model.BehaviorFlag_STAT_DEF_DOWN,
+		},
+		Stacking:          modifier.ReplaceBySource,
+		StatusType:        model.StatusType_STATUS_BUFF,
+		MaxCount:          4,
+		CountAddWhenStack: 1,
+		Listeners: modifier.Listeners{
+			OnAdd: recalcDefShred,
```

Comment:
change this to `OnAfterHit`, such as in here: https://github.com/simimpact/srsim/blob/59d22f4a344da43f147fb273d95c66e2496599e6/internal/lightcone/hunt/sleeplikethedead/sleeplikethedead.go#L32-L35
this way, you can check if the attack type is DoT before adding the DEF shred